external-crates/move/tools/move-cli/src/base/new.rs

```diff
@@ -76,6 +95,8 @@ version = \"{version}\"
 [addresses]"
         )?;
         for (addr_name, addr_val) in addrs {
+            let addr_name = addr_name.to_string();
+            let addr_name = addr_name.trim().replace('-', "_");
```

Comment:
Feels a bit strange. Should we just be requiring this and panicking when it does not match?

Reply:
Not sure I understand what you mean by requiring this and panicking. My understanding of the requirement suggested by @damirka was that we don't want any hyphens in the package names and address names.

Reply:
But these address names (as far as I can tell) are not supplied by the user, only by the tool itself. So I don't think there should be anything to do?

Reply:
Do we maybe do something with them just on the Sui side?

Reply:
My understanding is this will either need to be done on the Sui side or here, and I personally would have a preference for here since it's a more general Move restriction and not a Sui specific restriction. 

I think we can get into this type of situation if the user does something like `sui move new my-package` then the `addr` being passed in will be `my-package` which will need to be handled here.